src/autocomplete/powershell.ts

```diff
@@ -0,0 +1,455 @@
+import * as util from 'util'
+import {EOL} from 'os'
+import {Config, Interfaces, Command} from '@oclif/core'
+
+function sanitizeSummary(description?: string): string {
+  if (description === undefined) {
+    // PowerShell:
+    // [System.Management.Automation.CompletionResult] will error out if will error out if you pass in an empty string for the summary.
+    return ' '
+  }
+  return description
+  .replace(/"/g, '""') // escape double quotes.
+  .replace(/`/g, '``') // escape backticks.
+  .split(EOL)[0] // only use the first line
+}
+
+type CommandCompletion = {
+  id: string;
+  summary: string;
+  flags: CommandFlags;
+};
+
+type CommandFlags = {
+  [name: string]: Command.Flag.Cached;
+};
+
+type Topic = {
+  name: string;
+  description: string;
+};
+
+export default class PowerShellComp {
+  protected config: Config;
+
+  private topics: Topic[];
+
+  private _coTopics?: string[];
+
+  private commands: CommandCompletion[];
+
+  constructor(config: Config) {
+    this.config = config
+    this.topics = this.getTopics()
+    this.commands = this.getCommands()
+  }
+
+  private get coTopics(): string[] {
+    if (this._coTopics) return this._coTopics
+
+    const coTopics: string[] = []
+
+    for (const topic of this.topics) {
+      for (const cmd of this.commands) {
+        if (topic.name === cmd.id) {
+          coTopics.push(topic.name)
+        }
+      }
+    }
+
+    this._coTopics = coTopics
+
+    return this._coTopics
+  }
+
+  private genCmdHashtable(cmd: CommandCompletion): string {
```

Comment:
generates a powershell hashtable for a command.

command ID: "doctor"
output:
```powershell
"doctor" = @{
"_command" = @{
  "summary" = "Gather CLI configuration data and run diagnostic tests to discover and report potential problems in your environment."
  "flags" = @{
    "help" = @{ "summary" = "Show help for command" }
    "json" = @{ "summary" = "Format output as json." }
    "command" = @{ "summary" = "Command to run in debug mode; results are written to a log file." }
    "plugin" = @{ "summary" = "Specific plugin on which to run diagnostics." }
    "output-dir" = @{ "summary" = "Directory to save all created files rather than the current working directory." }
    "create-issue" = @{ "summary" = "Create a new issue on our GitHub repo and attach all diagnostic results." }
  }
}
}

```